Bottom: 1c26e2db63c02bf6aa1e1391dcdf9405ec573254
Top:    1cf63fe68290c03711ca9f2ee0d393ad3932485b
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2014-06-27 04:09:58 +0100

Refresh of add-soldermask-layers

---

diff --git a/src/create.c b/src/create.c
index 982c3a3..6b0865d 100644
--- a/src/create.c
+++ b/src/create.c
@@ -119,6 +119,7 @@ pcb_colors_from_settings (PCBType *ptr)
   ptr->ViaSelectedColor = Settings.ViaSelectedColor;
   ptr->WarnColor = Settings.WarnColor;
   ptr->MaskColor = Settings.MaskColor;
+  ptr->MaskSelectedColor = Settings.MaskSelectedColor;
   for (i = 0; i < MAX_LAYER; i++)
     {
       ptr->Data->Layer[i].Color = Settings.LayerColor[i];
@@ -134,6 +135,11 @@ pcb_colors_from_settings (PCBType *ptr)
     Settings.ElementColor : Settings.InvisibleObjectsColor;
   ptr->Data->Layer[bottom_silk_layer].SelectedColor =
     Settings.ElementSelectedColor;
+
+  ptr->Data->Layer[top_soldermask_layer].Color = Settings.MaskColor;
+  ptr->Data->Layer[top_soldermask_layer].SelectedColor = Settings.MaskSelectedColor;
+  ptr->Data->Layer[bottom_soldermask_layer].Color = Settings.MaskColor;
+  ptr->Data->Layer[bottom_soldermask_layer].SelectedColor = Settings.MaskSelectedColor;
 }
 
 /* ---------------------------------------------------------------------------
diff --git a/src/global.h b/src/global.h
index f5cc05e..99b67d9 100644
--- a/src/global.h
+++ b/src/global.h
@@ -502,7 +502,7 @@ typedef struct PCBType
     *InvisibleObjectsColor,
     *InvisibleMarkColor,
     *ElementSelectedColor,
-    *RatSelectedColor, *ConnectedColor, *FoundColor, *WarnColor, *MaskColor;
+    *RatSelectedColor, *ConnectedColor, *FoundColor, *WarnColor, *MaskColor, *MaskSelectedColor;
   long CursorX,			/* cursor position as saved with layout */
     CursorY, Clipping;
   Coord Bloat,			/* drc sizes saved with layout */
@@ -637,7 +637,7 @@ typedef struct			/* some resources... */
     *OffLimitColor,
     *GridColor,
     *LayerColor[MAX_LAYER],
-    *LayerSelectedColor[MAX_LAYER], *WarnColor, *MaskColor;
+    *LayerSelectedColor[MAX_LAYER], *WarnColor, *MaskColor, *MaskSelectedColor;
   Coord ViaThickness,		/* some preset values */
     ViaDrillingHole, LineThickness, RatThickness, Keepaway,	/* default size of a new layout */
     MaxWidth, MaxHeight,
diff --git a/src/main.c b/src/main.c
index 9521397..26334b2 100644
--- a/src/main.c
+++ b/src/main.c
@@ -861,6 +861,15 @@ Color of the mask layer. Default value is @code{"#ff0000"}
 */
   COLOR (MaskColor, "#ff0000", "mask-color", "color for solder mask"),
 
+/* %start-doc options "3 Colors"
+@ftable @code
+@item --mask-color <string>
+Color of the mask layer. Default value is @code{"#ff0000"}
+@end ftable
+%end-doc
+*/
+  COLOR (MaskSelectedColor, "#00ffff", "mask-selected-color", "color of selected solder mask cutouts"),
+
 
 /* %start-doc options "5 Sizes"
 All parameters should be given with an unit. If no unit is given, 1/100 mil
